solve #3604

pranavr2003 opened the pull request:
There's an issue with formerly lineno 365, where streamlit *checks* if the SVG code starts with a `<svg`. This is obviously not the case with most SVG files, and hence it spits out a `FileNotFoundError` when the file starts with anything other than `<svg`. This is solved when the `if` statement that checks for `startswith()` is removed, so the SVG file is rendered regardless of  the starting tag(s)/strings.

pranavr2003 committed:
solve #3604

There's an issue with formerly lineno 365, where streamlit *checks* if the SVG code starts with a `<svg`. This is obviously not the case with most SVG files, and hence it spits out a `FileNotFoundError` when the file starts with anything other than `<svg`. This is solved when the `if` statement that checks for `startswith()` is removed, so the SVG file is rendered regardless of  the starting tag(s)/strings.

vdonato reviewed (changes requested):
Hi @pranavr2003, thanks for your contribution!

Before this is good to go, we'll need to address a few things.
* Some tests are failing due to formatting issues (I believe it's because the commented out code should be removed entirely rather than just commented out)
* It seems like this change broke some cases where an svg string is passed as an argument. See the failing e2e tests in the `cypress` pipeline of the CI build. I haven't looked too deeply into what's going on here, but whatever's breaking will need to be resolved before this PR is ready to be merged

pranavr2003 committed:
updated as per @vdonato's suggestion

pranavr2003 reviewed (commented):
@vdonato I've updated the PR as per your suggestion

pranavr2003 commented:
@vdonato Just checked and you're right. Passing in an SVG string indeed does not seem to work with this PR. So solving the problem of local SVG rendering is creating a problem of not rendering an SVG string. While removing the doctring part saying SVG strings can be passed can be done, I don't believe (and I'm sure you don't too) that this is the right way to handle this. I'll look into this in more detail. Let me know your thoughts.

pranavr2003 committed:
all test cases (manual-only) work now

All forms of SVG files (with/without tags, raw SVG code within `st.image()`) work.

pranavr2003 commented:
@vdonato Putting in a local SVG file with/without top level tags work now. Putting in raw SVG code (an SVG string) with/without top level tags works too. Please review and let me know your thoughts.

pranavr2003 commented:
@vdonato Test cases show local PNG and JPG files don't render now, with this PR. I'm not very experienced, unfortunately, so could you guide me through this and tell me what's going wrong and where the issue is?
